tests-e2e/lists_test.go

```diff
@@ -856,3 +856,121 @@ func TestMultipleSameList(t *testing.T) {
 		m.MatchList("2", m.MatchV3Count(16), m.MatchV3Ops()),
 	)
 }
+
+// NB: assumes bump_event_types is sticky
+func TestBumpEventTypesHandling(t *testing.T) {
+	alice := registerNewUser(t)
+	bob := registerNewUser(t)
+	charlie := registerNewUser(t)
+
+	t.Log("Alice creates two rooms")
+	room1 := alice.CreateRoom(
+		t,
+		map[string]interface{}{
+			"preset": "public_chat",
+			"name":   "room1",
+		},
+	)
+	room2 := alice.CreateRoom(
+		t,
+		map[string]interface{}{
+			"preset": "public_chat",
+			"name":   "room2",
+		},
+	)
+	t.Logf("room1=%s room2=%s", room1, room2)
+	t.Log("Bob joins both rooms.")
+	bob.JoinRoom(t, room1, nil)
+	bob.JoinRoom(t, room2, nil)
+
+	t.Log("Bob sends a message in room 2 then room 1.")
+	bob.SendEventSynced(t, room2, Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"body":    "Hi room 2",
+			"msgtype": "m.text",
+		},
+	})
+	bob.SendEventSynced(t, room1, Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"body":    "Hello world",
+			"msgtype": "m.text",
+		},
+	})
+
+	t.Log("Alice requests a sliding sync that bumps rooms on messages only.")
+	aliceReqList := sync3.RequestList{
+		Sort:   []string{sync3.SortByRecency, sync3.SortByName},
+		Ranges: sync3.SliceRanges{{0, 20}},
+		RoomSubscription: sync3.RoomSubscription{
+			RequiredState: [][2]string{{"m.room.avatar", ""}, {"m.room.encryption", ""}},
+			TimelineLimit: 10,
+		},
+	}
+	aliceSyncRequest := sync3.Request{
+		Lists: map[string]sync3.RequestList{
+			"room_list": aliceReqList,
+		},
+		BumpEventTypes: []string{"m.room.message", "m.room.encrypted"},
+	}
+	// TODO: to avoid flakes, want this sync to include/cover Bob's messages in both rooms.
+	aliceRes := alice.SlidingSync(t, aliceSyncRequest)
```

Comment:
It will because of how sliding sync works. It's lazy, so the first time you call sliding sync it will do an initial sync, which will return the most recent event in each room. This will be the two hello world messages, so there's no race conditions here, but only because the first SS request blocks on upstream /sync.